src/main/java/hudson/plugins/git/UserMergeOptions.java

```diff
@@ -64,7 +68,7 @@ public UserMergeOptions(String mergeTarget) {
      * @param pbm pre-build merge options used to construct UserMergeOptions
      */
     public UserMergeOptions(PreBuildMergeOptions pbm) {
-        this(pbm.getRemoteBranchName(), pbm.getMergeTarget(), pbm.getMergeStrategy().toString(), pbm.getFastForwardMode());
+        this(pbm.getRemoteBranchName(), pbm.getMergeTarget(), pbm.getMergeStrategy().toString(), pbm.getFastForwardMode(), pbm.getTimeout());
```

Comment:
By addint this timeout here does  the `preBuildMerge` Command automatically get decorated with the timeout ?

Reply:
An automated test should be written which asserts that the timeout is correctly decorated onto the git command.  The timeout value is written on the end of the logged git command as a comment.  It looks like '# timeout=XX'.  Refer to [CheckoutOptionWorkflowTest](https://github.com/jenkinsci/git-plugin/blob/7449005233974f715fa6ec6ee722d7ef755f6334/src/test/java/hudson/plugins/git/extensions/impl/CheckoutOptionWorkflowTest.java#L30) for one example.